src/Tearing/TearingEngine.inl

```diff
@@ -894,96 +903,73 @@ void TearingEngine<DataTypes>::handleEvent(sofa::core::objectmodel::Event* event
         {
             fractureSegmentEndpoints.push_back(Pb);
             fractureSegmentEndpoints.push_back(Pc);
-
         }
     }
-  
-    if (/* simulation::AnimateBeginEvent* ev = */simulation::AnimateEndEvent::checkEventType(event))
+
+    // Perform fracture every d_stepModulo
+    int step = d_stepModulo.getValue();
+    if (step == 0) // interactive version
     {
-        int step = d_stepModulo.getValue();
-        if (step == 0) // interactive version
-        {
-            if (m_stepCounter > 200 && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
-                algoFracturePath();
-        }
-        else if (((m_stepCounter % step) == 0) && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
-        {
-            if (m_stepCounter > d_stepModulo.getValue())
-                algoFracturePath();
-        }
+        if (m_stepCounter > 200 && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
+            algoFracturePath();
     }
-
-    if (sofa::core::objectmodel::KeypressedEvent* ev = dynamic_cast<sofa::core::objectmodel::KeypressedEvent*>(event))
+    else if (((m_stepCounter % step) == 0) && (m_tearingAlgo->getFractureNumber() < d_nbFractureMax.getValue()))
     {
-        if (ev->getKey() == 'C')
-        {
+        if (m_stepCounter > d_stepModulo.getValue())
             algoFracturePath();
-        }
     }
 }
 
+
 template <class DataTypes>
 void TearingEngine<DataTypes>::draw(const core::visual::VisualParams* vparams)
-{  
-    
+{     
+    const auto stateLifeCycle = vparams->drawTool()->makeStateLifeCycle();
```

Comment:
What do these new lines do exactly?

Reply:
you can check the code of this method in DrawTool.h but it makes sure OpenGL states are reset, when we exit this draw method, to the status they had before entering this method.
For example the default color is white in SOFA and if in this method we set glColor(red) for our drawing and forget to reset it to white, this will avoid all futur drawing from other method to be red.